### ISSUE TITLE
Completion with html doesn't work

### ISSUE THREAD

**Issue by krdzo** [emacs-lsp/lsp-mode#1449]
**Describe the bug**
Completion doesn't work in html or css buffers when the `lsp-enable-snippet` is set to `nil` on lsp-mode from melpa-stable. When using lsp-mode from the melpa repo it doesn't work at all.

**To Reproduce**
This is my init.el:
```
(ido-mode 1)
(setq ido-enable-flex-matching t)

;; package and use package setup
(require 'package)
(setq package-archives '(("org"       . "https://orgmode.org/elpa/")
                         ("gnu"       . "https://elpa.gnu.org/packages/")
                         ("melpa"     . "https://melpa.org/packages/")
                         ("melpa-stable" . "https://stable.melpa.org/packages/")))
(setq package-enable-at-startup nil)
(package-initialize)

(unless (package-installed-p 'use-package)
  (package-refresh-contents)
  (package-install 'use-package))
(eval-when-compile
  (require 'use-package))
(setq use-package-always-ensure t)

(use-package company
  :hook (after-init . global-company-mode)
  :config
  (setq company-minimum-prefix-length 1)
  (setq company-idle-delay 0.0)
  (setq company-show-numbers t)
  (setq company-selection-wrap-around t)
)

(use-package lsp-mode
  :pin melpa-stable  ;; this is here so that completion works at all
  :commands (lsp lsp-deferred)
  :init (setq lsp-keymap-prefix "C-c l")
  :hook ((html-mode . lsp)
         (css-mode . lsp))
  :config
  ;;(setq lsp-enable-snippet nil)
  )

(use-package company-lsp
  :commands company-lsp)
```
and this is the test.html
```
<!doctype html>
<html>
    <head>
	<title></title>
	<meta charset="utf-8" />
   </head>
    <body>
      
    </body>
</html>
```
First start emacs and start editing test.html. I get this warning from emacs

> Warning (emacs): Yasnippet is not installed, but ‘lsp-enable-snippet’ is set to ‘t’. You must either install yasnippet, or disable snippet support.

but the auto-completion in the test.html buffer is working.
company-diag shows:
>company-backends: (company-lsp company-bbdb company-eclim company-semantic company-clang company-xcode company-cmake company-files
	     (company-dabbrev-code company-gtags company-etags company-keywords)
	     company-oddmuse company-dabbrev)
>
>Used backend: company-lsp

Then in init.el uncomment the line `(setq lsp-enable-snippet nil)` and restart emacs. Begin to edit test.html and I don't get the warning but auto-completion doesn't work.
company-diag shows 
>company-backends: (company-lsp company-bbdb company-eclim company-semantic company-clang company-xcode company-cmake company-files
	     (company-dabbrev-code company-gtags company-etags company-keywords)
	     company-oddmuse company-dabbrev)
>
>Used backend: company-dabbrev

This is the difference in lsp-describe-session.
This is with (setq lsp-enable-snippets t), (the deffault):
```
 `-[-] html-ls:9001
    |-[-] Buffers
    |  `-[+] ar.html
    `-[-] Capabilities
       |-[X] foldingRangeProvider: t
       |-[X] colorProvider:
       |-[X] referencesProvider: t
       |-[-] signatureHelpProvider:
       |  `-[X] triggerCharacters: [(]
       |-[X] definitionProvider: t
       |-[X] documentSymbolProvider: t
       |-[-] documentLinkProvider:
       |  `-[X] resolveProvider: nil
       |-[X] documentRangeFormattingProvider: nil
       |-[X] documentHighlightProvider: t
       |-[X] hoverProvider: t
       |-[-] completionProvider:
       |  |-[X] triggerCharacters: [. : < " = /]
       |  `-[X] resolveProvider: t
       `-[X] textDocumentSync: 1
```

This is with (setq lsp-enable-snippets nil):
```
-[-] html-ls:9270
    |-[-] Buffers
    |  `-[+] ar.html
    `-[-] Capabilities
       |-[X] foldingRangeProvider: t
       |-[X] colorProvider:
       |-[X] referencesProvider: t
       |-[-] signatureHelpProvider:
       |  `-[X] triggerCharacters: [(]
       |-[X] definitionProvider: t
       |-[X] documentSymbolProvider: t
       |-[-] documentLinkProvider:
       |  `-[X] resolveProvider: nil
       |-[X] documentRangeFormattingProvider: nil
       |-[X] documentHighlightProvider: t
       |-[X] hoverProvider: t
       `-[X] textDocumentSync: 1
```
as you see there is no ` completionProvider` in the later.

**Expected behavior**
Expect to have completion in html/web buffers but if I change the lsp-enable-snippets variable to nill the completion doesn't work.

The same thing hapends in spacemacs if you change `lsp-enable-snippets` to `nil` under
`(defun dotspacemacs/user-config ())`

**Which Language Server did you use**
Using vscode-html-languageserver-bin and vscode-css-languageserver-bin

**OS**
Linux, Manjaro

**Comment by yyoncho** [comment]
This behaviour is caused by an lsp-mode improvement after the last release. If you don't have yasnippet we have to disregard the fact that lsp-enable-snippet is t and send nil because otherwise later we won't be able to expand the snippets. Apparently, the server disables the completion provider when the snippets are off, this is not something that can be fixed in lsp-mode itself but it should be addressed in the server which they probably won't because it is server extracted from vscode and they don't have that problem. So I would recommend you to install to yasnippet.

**Comment by yyoncho** [comment]
I am closing the bug because it seems like there is nothing to address on our side. Please reopen if you disagree.

**Comment by krdzo** [comment]
Ok. Just wanted to write this somewhere because I spent last 7 days around this. I making my own config for the first time and didn't have yasnippet installed yet, because I want to build the config gradually, so the completion didn't work. It was a lot of frustration and when I found out what the "bug" was I wanted to write it here so that someone who is in my shoes later on would find this and know what the hell is going on.

Thanks for the explanation and all the hard work you do.
